libraries/standard/mqtt/src/iot_mqtt_helper.c

```diff
@@ -353,14 +355,16 @@ bool _IotMqtt_ConnectPacketSize( const IotMqttConnectInfo_t * pConnectInfo,
     if( pConnectInfo->awsIotMqttMode == true )
     {
         #if AWS_IOT_MQTT_ENABLE_METRICS == 1
-            connectPacketSize += ( AWS_IOT_METRICS_USERNAME_LENGTH +
-                                   ( size_t ) ( pConnectInfo->userNameLength ) + sizeof( uint16_t ) );
-            encodedUserName = true;
+            if( ( ( size_t ) pConnectInfo->userNameLength + AWS_IOT_METRICS_USERNAME_LENGTH ) <= ( ( size_t ) ( UINT16_MAX ) ) )
+            {
+                connectPacketSize += ( AWS_IOT_METRICS_USERNAME_LENGTH +
+                                       ( size_t ) ( pConnectInfo->userNameLength ) + sizeof( uint16_t ) );
+
+                encodedUserName = true;
+            }
```

Comment:
Note that you're failing silently if the test at line 358 isn't true.  Shouldn't we log an error to let the user know that they have misconfigured something?  Same comment at line 163.

Reply:
This isn't really a misconfiguration.  We pass metrics to IoT core by appending a string to the username.  In the event appending a string to the username would cause the result to be longer than allowed, we don't pass metrics and just use the given username.

To the customer, it makes no difference if metrics are passed or not, as long as their username is given to the broker.